extensions/amp-consent/0.1/tcf-api-command-manager.js

```diff
@@ -78,12 +94,87 @@ export class TcfApiCommandManager {
         this.handleGetTcData_(payload, win);
         break;
       case TCF_POST_MESSAGE_API_COMMANDS.ADD_EVENT_LISTENER:
+        this.handleAddEventListner_(payload, win);
+        break;
       case TCF_POST_MESSAGE_API_COMMANDS.REMOVE_EVENT_LISTENER:
+        this.handleRemoveEventListner_(payload, win);
+        break;
       default:
         return;
     }
   }
 
+  /**
+   * Add a entry to our changeListeners to signify that there
+   * is another iframe intrested in listening for TCData changes.
+   *
+   * Each entry has a unique `listenerId` that will be sent
+   * back to the 3p iframe.
+   * @param {!Object} payload
+   * @param {!Window} win
+   */
+  handleAddEventListner_(payload, win) {
+    const listenerId = this.listenerId_;
+    if (!this.changeListeners_[listenerId]) {
+      this.changeListeners_[listenerId] = {
+        payload,
+        win,
+      };
+    }
+    this.listenerId_++;
+  }
+
+  /**
+   * @param {!Object} payload
+   * @param {!Window} win
+   */
+  handleRemoveEventListner_(payload, win) {
+    const {callId, parameter} = payload;
+    const success = !!this.changeListeners_[parameter];
+    if (success) {
+      delete this.changeListeners_[parameter];
+    }
+
+    this.sendTcfApiReturn_(win, /** returnValue */ undefined, callId, success);
+  }
+
+  /**
+   * Handler for when policy manager signal potential TCData
+   * change. Only triggers new TCData to be sent if TC String
+   * has been change and is non-null.
+   */
+  handleTcDataChange_() {
+    if (!Object.keys(this.changeListeners_).length) {
+      return;
+    }
+
+    this.getTcDataPromises_().then((consentPromises) => {
+      const newTcString = consentPromises[2];
+      if (!newTcString || newTcString === this.currentTcString_) {
+        return;
+      }
+
+      this.currentTcString_ = newTcString;
+      const listenerIds = Object.keys(this.changeListeners_);
+      for (let i = 0; i < listenerIds.length; i++) {
+        const listenerId = parseInt(listenerIds[i], 10);
```

Comment:
nit: FYI the second arg to `parseInt` is optional. Defaults to base 10. Also, IMHO `Number(someString)` works better.

Reply:
I actually think we have a presubmit check that forbids using parseInt w/out radix because of weird behavior